airflow-core/src/airflow/jobs/scheduler_job_runner.py

```diff
@@ -1815,9 +1816,10 @@ def _create_dag_runs(self, dag_models: Collection[DagModel], session: Session) -
             # we need to set DagModel.next_dagrun_info if the DagRun already exists or if we
             # create a new one. This is so that in the next scheduling loop we try to create new runs
             # instead of falling in a loop of IntegrityError.
```

Comment:
This isn't true anymore with the changes.

Reply:
Would that cause a bug where a `DagRun` exists but `DagModel.next_dagrun` wasn't updated (the edge case mentioned in the comment), the scheduler will get stuck in an infinite loop attempting to create the same dag run repeatedly?